File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        result=[]
        n=len(nums)
        for i in range(n//2):
            print(i)
            min_elem=min(nums)
            max_elem=max(nums)
            result.append((min_elem + max_elem) / 2)
            nums.remove(min_elem)
            nums.remove(max_elem)

        return(min(result))

        

File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.cpp
class Solution {
public:
    double minimumAverage(vector<int>& nums) {
        double min = static_cast<double>(INT_MAX);
        std::sort(nums.begin(), nums.end());
        while (!nums.empty()) {
            double average = (nums.front() + nums.back()) / 2.0;
            min = std::min(min, average);
            nums.erase(nums.begin()); // remove the first element
            nums.pop_back(); // remove the last element
        }
        return min;
    }
};